### PR TITLE
python-sphinx: update to 7.3.7

### PR DESCRIPTION
**Pull request by pfusik**

### CODE REVIEW

File: mingw-w64-python-sphinx/PKGBUILD [hunk 1]
```diff
@@ -25,6 +25,7 @@ depends=("${MINGW_PACKAGE_PREFIX}-python-babel"
          "${MINGW_PACKAGE_PREFIX}-python-docutils"
          "${MINGW_PACKAGE_PREFIX}-python-imagesize"
          "${MINGW_PACKAGE_PREFIX}-python-jinja"
+         "${MINGW_PACKAGE_PREFIX}-python-myst-parser"
```
Comment (pfusik) [inline mingw-w64-python-sphinx/PKGBUILD:28]: `python-myst-parser` was already needed for 7.2.6-2, which is why I looked into this package.

Comment (pfusik) [reply mingw-w64-python-sphinx/PKGBUILD]: Sorry, false alarm. It's https://github.com/llvm/llvm-project/blob/bbe9119d9cb37662faafe7fe273e792b1b70145e/llvm/docs/conf.py#L35 that imports `myst-parser` when running sphinx.

File: mingw-w64-python-sphinx/PKGBUILD [hunk 2]
```diff
@@ -37,6 +37,7 @@ depends=("${MINGW_PACKAGE_PREFIX}-python-babel"
          "${MINGW_PACKAGE_PREFIX}-python-sphinxcontrib-qthelp"
          "${MINGW_PACKAGE_PREFIX}-python-sphinxcontrib-serializinghtml")
 makedepends=("${MINGW_PACKAGE_PREFIX}-python-build"
+             "${MINGW_PACKAGE_PREFIX}-python-defusedxml"
```
Comment (Biswa96) [inline mingw-w64-python-sphinx/PKGBUILD:40]: Why is this required?

Comment (pfusik) [reply mingw-w64-python-sphinx/PKGBUILD:40]:
```
ImportError while loading conftest 'C:/0/src/MINGW/mingw-w64-python-sphinx/src/python-build-MINGW64/tests/conftest.py'.
tests/conftest.py:14: in <module>
    from sphinx.testing.util import _clean_up_global_state
sphinx/testing/util.py:14: in <module>
    from defusedxml.ElementTree import parse as xml_parse
E   ModuleNotFoundError: No module named 'defusedxml'
```

Comment (Biswa96) [reply mingw-w64-python-sphinx/PKGBUILD:40]: So, that is for testing. If you want you can add that in checkdepends and run the tests. Otherwise, that should not be in makedepends. See archlinux pkgbuild as reference.

Comment (pfusik) [reply mingw-w64-python-sphinx/PKGBUILD]: Moved to `checkdepends`.
